src/ProjectTemplates/Shared/Project.cs

```diff
@@ -102,6 +102,21 @@ public class Project : IDisposable
             try
             {
                 Output.WriteLine("Acquired DotNetNewLock");
+                
+                if (Directory.Exists(TemplateOutputDir))
+                {
+                    Output.WriteLine($"Template directory already exists, deleting contents of {TemplateOutputDir}");
+                    var di = new DirectoryInfo(TemplateOutputDir);
```

Comment:
I think `Directory.Delete(TemplateOutputDir, recursive: true)` is equivalent to this.